Unknown CMake command "block"

### Company or project name

CH

### Operating system

Does not matter

### CMake version

$ cmake --version
cmake version 3.22.1

### Ninja version

Does not matter

### Compiler name and version

Does not matter

### Full cmake and/or ninja output

```
-- Adding contrib module grpc (configuring with grpc-cmake)
-- Adding contrib module llvm-project (configuring with llvm-project-cmake)
-- LLVM TARGETS TO BUILD X86
-- LLVM CMAKE CROSS COMPILING 0
CMake Error at contrib/llvm-project-cmake/CMakeLists.txt:152 (block):
  Unknown CMake command "block".


-- Configuring incomplete, errors occurred!
See also "/mnt/ch/ClickHouse/build/CMakeFiles/CMakeOutput.log".
See also "/mnt/ch/ClickHouse/build/CMakeFiles/CMakeError.log".
```

cmake [`Block` was added in 3.25](https://cmake.org/cmake/help/latest/command/block.html) but we are currently requiring only 3.20

https://github.com/ClickHouse/ClickHouse/blob/83f16db7b8452393fb49290913e3fae74932e9c0/contrib/llvm-project-cmake/CMakeLists.txt#L152-L162

https://github.com/ClickHouse/ClickHouse/blob/83f16db7b8452393fb49290913e3fae74932e9c0/CMakeLists.txt#L1


We can either increase `cmake_minimum_required` or remove the block usage.
cc @thevar1able

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.20)
+cmake_minimum_required(VERSION 3.25)
 
 project(ClickHouse LANGUAGES C CXX ASM)
 

